### PR TITLE
NoTask-Added new search endpoints

### PR DESCRIPTION
**Pull request by lgrzybowski**
There a endpoint which do not have test coverage and need to be added to provide confident coverage for test search.

### CODE REVIEW

File: lib/bwapi/client/projects/preview_search.rb [hunk 1]
```diff
@@ -6,8 +6,64 @@ module PreviewSearch
...
         # TODO: Add parameters documentation
-        def project_preview_search_mentions(project_id)
-          post "/projects/#{project_id}/previewsearch/mentions"
+        def project_preview_search_mentions(project_id, opts = {})
+          post "/projects/#{project_id}/previewsearch/mentions", opts
+        end
+
+        # Create a project preview search for author
+        #
+        # TODO: Add parameters documentation
```
Comment (jonathanchrisp) [inline lib/bwapi/client/projects/preview_search.rb]: We should really add the parameters, but one for @bw-lee or @benSlaughter or @SirCN to decide 😄

Comment (lgrzybowski) [inline lib/bwapi/client/projects/preview_search.rb]: As this is not public repo anymore we can always use swagger and you have one place to maintain than two

Comment (jonathanchrisp) [inline lib/bwapi/client/projects/preview_search.rb]: Hmm, I can see your reasoning here but its now no longer consistent with the rest of the codebase.

Comment (lgrzybowski) [inline lib/bwapi/client/projects/preview_search.rb]: I don't want to be picky but It's not the first time when people follow that approach. I'm not that 1st person which is doing it,  that way. If you use search there is 88 element where TODO is found, so it's not consistent from quite long time. Thats why as this repo is no more public we have our internal documentation which we can use when we need it.

Comment (bw-lee) [inline lib/bwapi/client/projects/preview_search.rb]: The missing parameters are mostly missing from a time when the pressure was on to get completion, so Jon and Ben had to quickly add endpoints.  We really should start adding them when new endpoints are added.  At some point there will be an exercise to tidy up and fix the missing option documentation.  Irrespective of whether the repo is public or private.

Comment (lgrzybowski) [inline lib/bwapi/client/projects/preview_search.rb]: :+1

**Comment by bw-lee** [comment]
LGTM

**Comment by jonathanchrisp** [comment]
@great job @lgrzybowski

**Comment by jonathanchrisp** [comment]
@benSlaughter @bw-lee I'll let you merge

**Comment by benSlaughter** [comment]
LGTM

**Comment by lgrzybowski** [comment]
I cannot merge this ... :( as I don't have right to do this, can some pls press this green button and add right to project that everyone can do this.

**Pull request merged**